sherpa_contrib/diag_resp.py

```diff
@@ -0,0 +1,712 @@
+                  subchan: int|None = None,
+                  chantype: str|None = None) -> tuple[npt.NDArray, npt.NDArray, int]:
+        """
+        return energy grid and channel offset for a given telescope
+        and instrument configuration
+        """
+
+        from sherpa.astro.io import backend as shp_backend
+
+        fn = f"{os.environ['ASCDS_INSTALL']}/data/ebounds-lut/{telescope.lower()}-ebounds-lut.fits"
+
+        chantype = self._set_chantype_none(telescope=telescope,
+                                           instrument=instrument,
+                                           chantype=chantype)
+
+        blkname = self._get_ebounds_blk(instrument=instrument,
+                                        detnam=detnam,
+                                        instfilter=instfilter,
+                                        subchan=subchan,
+                                        chantype=chantype)
+
+        try:
+            if "pyfits" in shp_backend.__name__:
+                from astropy.io import fits as aspy_fits
+
+                with aspy_fits.open(fn) as hdul:
+                    data = hdul[blkname].data
+
+                    chan = data.CHANNEL
+                    emin = data.E_MIN
+                    emax = data.E_MAX
+
+            else:
+                from pycrates import read_file as pcr_readfile
+
+                data = pcr_readfile(f"{fn}[{blkname}]")
+
+                chan = data.CHANNEL.values.astype(int)
+                emin = data.E_MIN.values.astype(float)
+                emax = data.E_MAX.values.astype(float)
+
+        except FileNotFoundError:
+            raise ValueError("The specified 'telescope' parameter value is invalid; no corresponding lookup table is available.")
+
+
+        offset = min(chan)
+
+        del(shp_backend)
+        del(data)
+
+        egrid_unit = get_keys_from_file(f"{fn}[{blkname}]")["EUNIT"]
+
+        if egrid_unit == "eV":
+            ## convert eV to keV ##
+            emin /=  1_000
+            emax /=  1_000
+
+        if egrid_unit == "MeV":
+            ## convert MeV to keV ##
+            emin *= 1_000
+            emax *= 1_000
+
+        if egrid_unit == "GeV":
+            ## convert MeV to keV ##
+            emin *= 1_000_000
+            emax *= 1_000_000
+
+        return emin, emax, offset
+
+
+
+def _check_and_update_instrument_info(telescope: str = "",
+                                      instrument: str|None = None,
+                                      detnam: str|None = None,
+                                      instfilter: str|None = None):
+    """
+    revise telescope/instrument configuration information to the
+    minimal parameters required to obtain the necessary energy grid
+    """
+
+    if telescope == "" or telescope is None:
+        raise ValueError("'telescope' parameter must be specified!")
+
+
+    ### update telescopes with alternate names ###
+    if telescope.upper() == "XTE":
```

Comment:
Instead of these cascading `if` statements, you might want to use a dictionary instead

```python
valid_vals = {'xte': 'RXTE', 'sax' : 'BeppoSAX', 'erosita': 'SRG'}
if telescope in valid_vals:
  telescope = valid_vals[telescope]
else:
  raise ValueError(...stuff...)


```

Reply:
addressed in commit bad862a80b9d2a79a7c5aa7718962c30d28dc5af